.github/workflows/version_check.yml

```diff
@@ -0,0 +1,198 @@
+        run: |
+          using Pkg
+          Pkg.activate(temp=true)
+          Pkg.add(["YAML", "TOML", "JSON", "HTTP"])
+          import YAML
+          import TOML
+          import JSON
+          import HTTP
+
+          PROJECT_TOML_PATH = "Project.toml"
+          QUARTO_YML_PATH = "_quarto.yml"
+          MANIFEST_TOML_PATH = "Manifest.toml"
+
+          function major_minor_match(vs...)
+              for v in vs
+                  if v.:major != vs[1].:major || v.:minor != vs[1].:minor
+                      return false
+                  end
+              end
+              return true
+          end
+
+          function major_minor_patch_match(vs...)
+              for v in vs
+                  if v.:major != vs[1].:major || v.:minor != vs[1].:minor || v.:patch != vs[1].:patch
+                      return false
+                  end
+              end
+              return true
+          end
+
+          function update_project_toml(fname, target_version)
+              # Don't deserialise/serialise as this will scramble lines
+              lines = readlines(fname)
+              open(fname, "w") do io
+                  for line in lines
+                      if occursin(r"^Turing\s*=\s*\"\d+\.\d+\"\s*$", line)
+                          println(io, "Turing = \"$(target_version.:major).$(target_version.:minor)\"")
+                      else
+                          println(io, line)
+                      end
+                  end
+              end
+          end
+
+          function update_quarto_yml(fname, target_version)
+              # Don't deserialise/serialise as this will scramble lines
+              lines = readlines(fname)
+              open(fname, "w") do io
+                  for line in lines
+                      m = match(r"^(\s+)- text:\s*\"v\d+\.\d+\"\s*$", line)
+                      if m !== nothing
+                          println(io, "$(m[1])- text: \"v$(target_version.:major).$(target_version.:minor)\"")
+                      else
+                          println(io, line)
+                      end
+                  end
+              end
+          end
+
+          # Retain the original version number string for error messages, as
+          # VersionNumber() will tack on a patch version of 0
+          quarto_yaml = YAML.load_file(QUARTO_YML_PATH)
+          quarto_version_str = quarto_yaml["website"]["navbar"]["right"][1]["text"]
+          quarto_version = VersionNumber(quarto_version_str)
+          println("_quarto.yml version: ", quarto_version_str)
+
+          project_toml = TOML.parsefile(PROJECT_TOML_PATH)
+          project_version_str = project_toml["compat"]["Turing"]
+          project_version = VersionNumber(project_version_str)
+          println("Project.toml version: ", project_version_str)
+
+          manifest_toml = TOML.parsefile(MANIFEST_TOML_PATH)
+          manifest_version = VersionNumber(manifest_toml["deps"]["Turing"][1]["version"])
```

Comment:
What does the `[1]` do? Does it presume some fixed ordering of things in the Manifest?

Reply:
I'm not fluent in TOML, but I think it's because every entry in the Manifest is specified like

```toml
[[deps.Turing]]
version = "whatever"
```

and the `[1]` comes from having to unpack the extra level of nesting with the double-square-brackets.

Reply:
I wonder if there's any case where this might not happen. Maybe if there are two packages named Turing but with different uuids?!

Reply:
Pkg refuses to add two packages with the same name (I just tried). No idea what purpose the extra level of nesting serves, then; but given that I failed to come up with an example pathological enough to break it, I no longer feel too bad about this magic `[1]`.

Reply:
Even if there is some case I'm not aware of when you might have duplicate entries, I don't think it's going to happen to the dependency called `Turing` in this repo. Happy with this.